### PR TITLE
Fix empty html report bug

### LINKED ISSUE #938: validate-traffic generating empty html report
Steps to repro:

1. git checkout main branch of azure-rest-api-specs
2. test-proxy restore -a specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/assets.json
3. mkdir .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic/
4. oav traffic-convert .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/recordings .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic/
5. oav validate-traffic .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json

_note_: if there's trouble in restoring test-proxy recording, you can download it directly: https://github.com/Azure/azure-sdk-assets/releases/tag/apitest%2Fappconfiguration%2Fmanagement_164ac7ad8e

Console outputs:
``` 
➜  azure-rest-api-specs git:(main) ✗ oav traffic-convert .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/recordings .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic/
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
{ level: '\x1B[31merror\x1B[39m' }
➜  azure-rest-api-specs git:(main) ✗ ls .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic/
basic0.json  basic10.json  basic12.json  basic14.json  basic16.json  basic18.json  basic2.json   basic21.json  basic23.json  basic25.json  basic27.json  basic29.json  basic30.json  basic32.json  basic34.json  basic36.json  basic38.json  basic4.json   basic41.json  basic43.json  basic45.json  basic47.json  basic49.json  basic50.json  basic52.json  basic7.json  basic9.json
basic1.json  basic11.json  basic13.json  basic15.json  basic17.json  basic19.json  basic20.json  basic22.json  basic24.json  basic26.json  basic28.json  basic3.json   basic31.json  basic33.json  basic35.json  basic37.json  basic39.json  basic40.json  basic42.json  basic44.json  basic46.json  basic48.json  basic5.json   basic51.json  basic6.json   basic8.json
➜  azure-rest-api-specs git:(main) ✗ oav validate-traffic .assets/OrAXPEIzPs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/scenarios/traffic specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Creating livevalidator with options:{"checkUnderFileRoot":false,"loadValidatorInBackground":false,"directory":"/home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01","swaggerPathsPattern":["appconfiguration.json"],"excludedSwaggerPathsPattern":["**/examples/**/*","**/quickstart-templates/**/*","**/schema/**/*","**/live/**/*","**/wire-format/**/*","**/azurestack/**/*"],"git":{"shouldClone":false,"url":"https://github.com/Azure/azure-rest-api-specs.git"},"swaggerPaths":[],"isPathCaseSensitive":false,"loadValidatorInInitialize":false,"isArmCall":false,"enableRoundTripLazyBuild":true}'
}
{ level: '\x1B[32minfo\x1B[39m', message: 'Get swagger path.' }
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Using swaggers found from directory: "/home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01" and pattern: "/home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json".\n' +
    '      Total paths count: 1'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Using swaggers found from directory: "/home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01" and pattern: "/home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json".\n' +
    '      Total paths count: 1'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Load spec /home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Add spec to cache /home/me/workspace/azure-rest-api-specs/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Complete loading spec appconfiguration.json'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Apply global transforms for all specs'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Cache initialization complete.'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Cache complete initialization with DurationInMs:143'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Fallback to microsoft.unknown -> unknown-api-version'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
{
  level: '\x1B[31merror\x1B[39m',
  message: 'ErrorMessage:Could not find provider "microsoft.unknown" in the cache..ErrorStack:undefined'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Fallback to microsoft.unknown -> unknown-api-version'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
{
  level: '\x1B[31merror\x1B[39m',
  message: 'ErrorMessage:Could not find provider "microsoft.unknown" in the cache..ErrorStack:undefined'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Fallback to microsoft.unknown -> unknown-api-version'
}
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.unknown" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%27)
}
Error: Undefined operation {"apiVersion":"unknown-api-version","operationId":"unknownOperationId","validationRequest":{"providerNamespace":"microsoft.unknown","resourceType":"/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7","apiVersion":"2020-06-01","requestMethod":"put","host":"management.azure.com","pathStr":"/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7","query":{"api-version":"2020-06-01"},"correlationId":"","activityId":"","requestUrl":"[https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01"}}](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourcegroups/apiTest-tyxep7?api-version=2020-06-01%22%7D%7D)
{
  level: '\x1B[32minfo\x1B[39m',
  message: 'Could not find provider "microsoft.network" in the cache. with requestUrl [https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourceGroups/apiTest-tyxep7/providers/Microsoft.Network/virtualNetworks/configstordeuje5?api-version=2022-07-01'](https://management.azure.com/subscriptions/db5eb68e-73e2-4fa8-b18a-46cd1be4cce5/resourceGroups/apiTest-tyxep7/providers/Microsoft.Network/virtualNetworks/configstordeuje5?api-version=2022-07-01%27)
}

```

The generated SwaggerAccuracyReport.html is empty:
image

### PR DESCRIPTION
**Pull request by leniatgh**
fixes #938

### CODE REVIEW

**Review by jordan13game — APPROVED** [review]

**Review by xiaoxuqi-ms — APPROVED** [review]

File: lib/liveValidation/liveValidator.ts [hunk 1]
```diff
@@ -993,21 +966,18 @@ export function formatUrlToExpectedFormat(requestUrl: string): string {
...
+ * @param headers Optional headers
  *
  * @returns parsed ValidationRequest info.
  */
 export const parseValidationRequest = (
   requestUrl: string,
-  requestMethod: string | undefined | null,
-  correlationId: string,
-  activityId: string
+  requestMethod: string,
```
Comment (raych1) [inline lib/liveValidation/liveValidator.ts]: Can you revert type change which will break current usage?

File: lib/commands/traffic-converter.ts [hunk 2]
```diff
@@ -73,10 +73,12 @@ function requestUriConversion(uri: string, version: string): string {
 }
 
 function processFile(file: string, inputJson: any) {
-  const filePrefix = file.substring(0, file.lastIndexOf("."));
   if (inputJson.Entries !== undefined && inputJson.Entries.length > 0) {
     inputJson.Entries.forEach((entry: ProxyPayload, idx: number) => {
-      let outFile = filePrefix + idx + ".json";
+      let outFile = `${file.substring(0, file.lastIndexOf("."))}_${String(idx).padStart(
```
Comment (raych1) [inline lib/commands/traffic-converter.ts]: filePrefix is set once out of foreach loop, why do we move it?

File: lib/swaggerValidator/trafficValidator.ts [hunk 3]
```diff
@@ -146,7 +145,9 @@ export class TrafficValidator {
         spec = await this.loader.load(pathResolve(swaggerPath));
```
Comment (raych1) [inline lib/swaggerValidator/trafficValidator.ts:145]: Can you remove `pathResolve` here as dup?

File: lib/swaggerValidator/trafficValidator.ts [hunk 4]
```diff
@@ -213,75 +208,75 @@ export class TrafficValidator {
...
-          if (validationResult.requestValidationResult.isSuccessful) {
-            liveRequestResponseList = findPathsToKey({ key: "liveResponse", obj: trafficSpec });
-          } else {
-            liveRequestResponseList = findPathsToKey({ key: "liveRequest", obj: trafficSpec });
-          }
-          const liveRequestResponsePosition = getFilePositionFromJsonPath(
-            trafficSpec,
-            liveRequestResponseList[0]
-          );
+        const trafficSpec = await this.swaggerLoader.load(payloadFilePath);
```
Comment (raych1) [inline lib/swaggerValidator/trafficValidator.ts:211]: Can you explain why if condition of validation failure on line#216 is removed?

**Review by raych1 — CHANGES_REQUESTED** [review]
few comments.

File: lib/swaggerValidator/trafficValidator.ts (hunk 3 above)
Comment (leniatgh) [reply lib/swaggerValidator/trafficValidator.ts:145]: this is from liveValidator.swaggerList, different from `specPath`

File: lib/liveValidation/liveValidator.ts (hunk 1 above)
Comment (leniatgh) [reply lib/liveValidation/liveValidator.ts]: okay, I will restore the legacy one. After the usage in live validation service is updated, we can remove it then.

File: lib/swaggerValidator/trafficValidator.ts (hunk 4 above)
Comment (leniatgh) [reply lib/swaggerValidator/trafficValidator.ts:211]: operation without errors should also be counted as covered operations.

File: lib/commands/traffic-converter.ts (hunk 2 above)
Comment (leniatgh) [reply lib/commands/traffic-converter.ts]: good catch, fixed.

File: lib/liveValidation/liveValidator.ts (hunk 1 above)
Comment (leniatgh) [reply lib/liveValidation/liveValidator.ts]: fixed, the exported name in index.ts is kept same as before.

**Review by raych1 — APPROVED** [review]

**Pull request merged**